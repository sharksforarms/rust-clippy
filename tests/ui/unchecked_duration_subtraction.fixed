#![warn(clippy::unchecked_duration_subtraction)]

use std::time::{Duration, Instant};

fn main() {
    let instant = Instant::now();
    let duration = Duration::from_secs(3);
    let duration2 = Duration::from_secs(1);

    let _ = instant.checked_sub(duration).unwrap();

    let _ = Instant::now().checked_sub(Duration::from_secs(5)).unwrap();

    let _ = instant.checked_sub(Duration::from_secs(5)).unwrap();

    let _ = Instant::now().checked_sub(duration).unwrap();

    let _ = Duration::from_secs(1).checked_sub(duration).unwrap();

    let _ = duration2.checked_sub(duration).unwrap();

    let _ = Instant::now().elapsed().checked_sub(duration).unwrap();
}
