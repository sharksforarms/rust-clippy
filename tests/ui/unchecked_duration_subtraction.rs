#![warn(clippy::unchecked_duration_subtraction)]

use std::time::{Duration, Instant};

fn main() {
    let instant = Instant::now();
    let duration = Duration::from_secs(3);
    let duration2 = Duration::from_secs(1);

    let _ = instant - duration;

    let _ = Instant::now() - Duration::from_secs(5);

    let _ = instant - Duration::from_secs(5);

    let _ = Instant::now() - duration;

    let _ = Duration::from_secs(1) - duration;

    let _ = duration2 - duration;

    let _ = Instant::now().elapsed() - duration;
}
