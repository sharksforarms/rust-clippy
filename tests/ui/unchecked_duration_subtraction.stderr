error: unchecked subtraction of 'Duration' from 'Instant'
  --> tests/ui/unchecked_duration_subtraction.rs:10:13
   |
LL |     let _ = instant - duration;
   |             ^^^^^^^^^^^^^^^^^^ help: try: `instant.checked_sub(duration).unwrap()`
   |
   = note: `-D clippy::unchecked-duration-subtraction` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unchecked_duration_subtraction)]`

error: unchecked subtraction of 'Duration' from 'Instant'
  --> tests/ui/unchecked_duration_subtraction.rs:12:13
   |
LL |     let _ = Instant::now() - Duration::from_secs(5);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Instant::now().checked_sub(Duration::from_secs(5)).unwrap()`

error: unchecked subtraction of 'Duration' from 'Instant'
  --> tests/ui/unchecked_duration_subtraction.rs:14:13
   |
LL |     let _ = instant - Duration::from_secs(5);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `instant.checked_sub(Duration::from_secs(5)).unwrap()`

error: unchecked subtraction of 'Duration' from 'Instant'
  --> tests/ui/unchecked_duration_subtraction.rs:16:13
   |
LL |     let _ = Instant::now() - duration;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Instant::now().checked_sub(duration).unwrap()`

error: unchecked subtraction of 'Duration' from 'Duration'
  --> tests/ui/unchecked_duration_subtraction.rs:18:13
   |
LL |     let _ = Duration::from_secs(1) - duration;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Duration::from_secs(1).checked_sub(duration).unwrap()`

error: unchecked subtraction of 'Duration' from 'Duration'
  --> tests/ui/unchecked_duration_subtraction.rs:20:13
   |
LL |     let _ = duration2 - duration;
   |             ^^^^^^^^^^^^^^^^^^^^ help: try: `duration2.checked_sub(duration).unwrap()`

error: unchecked subtraction of 'Duration' from 'Duration'
  --> tests/ui/unchecked_duration_subtraction.rs:22:13
   |
LL |     let _ = Instant::now().elapsed() - duration;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Instant::now().elapsed().checked_sub(duration).unwrap()`

error: aborting due to 7 previous errors

